src/react/useQueries.ts

```diff
@@ -20,18 +22,30 @@ export function useQueries(queries: UseQueryOptions[]): UseQueryResult[] {
     return defaultedOptions
   })
 
-  const [observer] = React.useState(
-    () => new QueriesObserver(queryClient, defaultedQueries)
+  const observerRef = React.useRef(
+    new QueriesObserver(queryClient, defaultedQueries)
   )
 
-  const result = observer.getOptimisticResult(defaultedQueries)
+  const result = observerRef.current.getOptimisticResult(defaultedQueries)
+
+  React.useEffect(() => {
+    // Do not notify on updates because of changes in the options because
+    // these changes should already be reflected in the optimistic result.
+    observerRef.current.setQueries(defaultedQueries, { listeners: false })
+  }, [defaultedQueries])
+
+  const someSuspense = defaultedQueries.some(q => q.suspense)
+  const someUseErrorBoundary = defaultedQueries.some(q => q.useErrorBoundary)
+  const firstResultWithError = result.find(r => r.error)
+  const someError = firstResultWithError?.error
+  const someIsLoading = result.some(r => r.isLoading)
 
   React.useEffect(() => {
     mountedRef.current = true
 
-    const unsubscribe = observer.subscribe(
+    const unsubscribe = observerRef.current.subscribe(
       notifyManager.batchCalls(() => {
-        if (mountedRef.current) {
+        if (mountedRef.current && someIsLoading) {
```

Comment:
why do we need this check for `someIsLoading` here please? This looks like we're never force-updating if nothing is loading, which doesn't seem right 🤔